### PR TITLE
Dev s100/feature/188105 publish unitofsaleupdated event if no actions generated

### PR DESCRIPTION
**Pull request by Tukaram96**
Hi,

Dependency: - #259 

This PR contains code changes, unit tests of publish unitofsaleupdated event if no actions generated.
Please review.

Regards!

### CODE REVIEW

**Review by gaurav15233 — APPROVED** [review]
LGTM

**Review by jagannathsankpal — APPROVED** [review]
LGTM

File: src/UKHO.ERPFacade.API/Handlers/S100EventHandler.cs [hunk 1]
```diff
@@ -25,20 +27,22 @@ public class S100EventHandler : IEventHandler
         private readonly IBaseXmlTransformer _baseXmlTransformer;
         private readonly ISapClient _sapClient;
         private readonly IOptions<SapConfiguration> _sapConfig;
-
+        private readonly IS100UnitOfSaleUpdatedEventPublishingService _s100UnitOfSaleUpdatedEventPublishingService;
```
Comment (Vishal14583) [inline src/UKHO.ERPFacade.API/Handlers/S100EventHandler.cs:30]: Add a new line after this.

File: src/UKHO.ERPFacade.API/Handlers/S100EventHandler.cs [hunk 2]
```diff
@@ -71,17 +75,30 @@ public async Task ProcessEventAsync(BaseCloudEvent baseCloudEvent)
 
             _logger.LogInformation(EventIds.S100EventXMLStoredInAzureBlobContainer.ToEventId(), "S-100 data content published event xml payload is stored in azure blob container.");
 
-            var response = await _sapClient.PostEventData(sapPayload, _sapConfig.Value.SapEndpointForS100Event, _sapConfig.Value.SapServiceOperationForS100Event, _sapConfig.Value.SapUsernameForS100Event, _sapConfig.Value.SapPasswordForS100Event);
-
-            if (!response.IsSuccessStatusCode)
+            if (sapPayload.DocumentElement != null && int.TryParse(sapPayload.SelectSingleNode(XmlTemplateInfo.XpathNoOfActions).InnerText, out int actionCount) && actionCount <= 0)
             {
-                throw new ERPFacadeException(EventIds.S100RequestToSapFailedException.ToEventId(), $"An error occurred while sending S-100 data content to SAP. | {response.StatusCode}");
+                Result result = await _s100UnitOfSaleUpdatedEventPublishingService.PublishEvent(baseCloudEvent, s100EventData.CorrelationId);
```
Comment (Vishal14583) [inline src/UKHO.ERPFacade.API/Handlers/S100EventHandler.cs:80]:
```suggestion
                var result = await _s100UnitOfSaleUpdatedEventPublishingService.PublishEvent(baseCloudEvent, s100EventData.CorrelationId);
```

File: tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs [hunk 3]
```diff
@@ -110,5 +111,64 @@ public void WhenSapRespondsWith401Unauthorized_ThenS100EventHandlerRaiseAnExcept
...
         }
+
+        [Test]
+        public void WhenValidEventReceivedWithNoSAPAction_ThenS100EventHandlerPublishTheUnitOfsaleUpdatedEventToEesSuccessfully()
+        {
+            var fakeS100EventDataJson = JObject.Parse(@"{""data"":{""correlationId"":""123""}}");
+            var fakeS100EventData = JsonConvert.DeserializeObject<BaseCloudEvent>(fakeS100EventDataJson.ToString());
+
+            XmlDocument xmlDocument = new XmlDocument();
+            xmlDocument.LoadXml("<NOOFACTIONS>0</NOOFACTIONS>");
```
Comment (Vishal14583) [inline tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs:122]: Can you add a xml file for this.

File: tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs [hunk 4]
```diff
@@ -110,5 +111,64 @@ public void WhenSapRespondsWith401Unauthorized_ThenS100EventHandlerRaiseAnExcept
 
             A.CallTo(() => _fakeAzureTableReaderWriter.UpdateEntityAsync(A<string>.Ignored, A<string>.Ignored, A<Dictionary<string, object>>.Ignored)).MustNotHaveHappened();
         }
+
+        [Test]
+        public void WhenValidEventReceivedWithNoSAPAction_ThenS100EventHandlerPublishTheUnitOfsaleUpdatedEventToEesSuccessfully()
```
Comment (Vishal14583) [inline tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs:116]: Asserts for PostEventData and UpdateEntityAsync is missing.

**Review by Vishal14583 — CHANGES_REQUESTED** [review]
Few minor review comments

File: src/UKHO.ERPFacade.API/Handlers/S100EventHandler.cs (hunk 2 above)
Comment (Tukaram96) [reply src/UKHO.ERPFacade.API/Handlers/S100EventHandler.cs]: resolved as suggested.

File: tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs (hunk 3 above)
Comment (Tukaram96) [reply tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs]: resolved as suggested.

File: tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs (hunk 4 above)
Comment (Tukaram96) [reply tests/UKHO.ERPFacade.API.UnitTests/Handlers/S100EventHandlerTests.cs:116]: resolved as suggested.

**Review by Vishal14583 — APPROVED** [review]
LGTM

**Review by rushdynajath — APPROVED** [review]
looks good

**Pull request merged**